core/src/main/java/org/fao/geonet/api/records/attachments/CMISStore.java

```diff
@@ -426,6 +436,50 @@ public MetadataResourceContainer getResourceContainerDescription(final ServiceCo
         }
     }
 
+    @Override
+    public void copyResources(ServiceContext context, String sourceUuid, String targetUuid, MetadataResourceVisibility metadataResourceVisibility, boolean sourceApproved, boolean targetApproved) throws Exception {
+        final int sourceMetadataId = canEdit(context, sourceUuid, metadataResourceVisibility, sourceApproved);
+        final String sourceResourceTypeDir = getMetadataDir(context, sourceMetadataId) + cmisConfiguration.getFolderDelimiter() + metadataResourceVisibility.toString();
+        try {
+            Folder sourceParentFolder = (Folder) cmisConfiguration.getClient().getObjectByPath(sourceResourceTypeDir);
+
+            Map<String, Document> sourceDocumentMap = cmisUtils.getCmisObjectMap(sourceParentFolder, null);
+
+
+            for (Map.Entry<String, Document> sourceEntry : sourceDocumentMap.entrySet()) {
+                Document sourceDocument = sourceEntry.getValue();
+                if (sourceDocument instanceof Document) {
```

Comment:
I don't want to put too much constraints to assume this is a document for sure. If it's not a document, then it will do just skip it and look for the next cmis object. This code will not affect any performance by just for more safety.

Anything can be possible happening at the backend content server or its script. We can do more thing maybe put a warning/error here to log it.

Reply:
I don't think you understand my point. 
The following identifies that sourceDocument is a "Document" object.
`Document sourceDocument`
So it is not possible for the following to fail the condition.
`if (sourceDocument instanceof Document) {`
I'm surprised that the compiler is not complaining about this by stating that the condition will always be true.

Reply:
oh. I get it. Thanks